### ISSUE TITLE
ltx-2 stuck Loading checkpoint shards

### ISSUE THREAD

**Issue by Tirnaia** [ostris/ai-toolkit#714]
Has anyone had the same problem?

![Image](https://github.com/user-attachments/assets/b412158d-bd6f-46f2-9c2f-f3e9282c8998)

**Comment by Tirnaia** [comment]
job: "extension"
config:
  name: "ltx2_prova"
  process:
    - type: "diffusion_trainer"
      training_folder: "/ai-toolkit/output"
      sqlite_db_path: "./aitk_db.db"
      device: "cuda"
      trigger_word: null
      performance_log_every: 10
      network:
        type: "lora"
        linear: 32
        linear_alpha: 32
        conv: 16
        conv_alpha: 16
        lokr_full_rank: true
        lokr_factor: -1
        network_kwargs:
          ignore_if_contains: []
      save:
        dtype: "bf16"
        save_every: 250
        max_step_saves_to_keep: 4
        save_format: "diffusers"
        push_to_hub: false
      datasets:
        - folder_path: "/ai-toolkit/datasets/whisk"
          mask_path: null
          mask_min_value: 0.1
          default_caption: ""
          caption_ext: "txt"
          caption_dropout_rate: 0.05
          cache_latents_to_disk: true
          is_reg: false
          network_weight: 1
          resolution:
            - 512
          controls: []
          shrink_video_to_frames: true
          num_frames: 1
          flip_x: false
          flip_y: false
          num_repeats: 1
          do_i2v: true
          fps: 24
      train:
        batch_size: 1
        bypass_guidance_embedding: false
        steps: 5000
        gradient_accumulation: 1
        train_unet: true
        train_text_encoder: false
        gradient_checkpointing: true
        noise_scheduler: "flowmatch"
        optimizer: "adamw8bit"
        timestep_type: "weighted"
        content_or_style: "balanced"
        optimizer_params:
          weight_decay: 0.0001
        unload_text_encoder: false
        cache_text_embeddings: true
        lr: 0.0001
        ema_config:
          use_ema: false
          ema_decay: 0.99
        skip_first_sample: false
        force_first_sample: false
        disable_sampling: false
        dtype: "bf16"
        diff_output_preservation: false
        diff_output_preservation_multiplier: 1
        diff_output_preservation_class: "person"
        switch_boundary_every: 1
        loss_type: "mse"
      logging:
        log_every: 1
        use_ui_logger: true
      model:
        name_or_path: "Lightricks/LTX-2"
        quantize: true
        qtype: "qfloat8"
        quantize_te: true
        qtype_te: "uint4"
        arch: "ltx2"
        low_vram: true
        model_kwargs: {}
        layer_offloading: true
        layer_offloading_text_encoder_percent: 0
        layer_offloading_transformer_percent: 1
      sample:
        sampler: "flowmatch"
        sample_every: 250
        width: 768
        height: 768
        samples:
          - prompt: " "
          - prompt: " "
        neg: ""
        seed: 42
        walk_seed: true
        guidance_scale: 4
        sample_steps: 25
        num_frames: 1
        fps: 1
meta:
  name: "[name]"
  version: "1.0"

**Comment by james-s-tayler** [comment]
It's not actually stuck, it's just going extremely slow and basically writing everything to the swapfile. If you only have 16GB VRAM and 64GB system RAM then it uses around 25GB of swap in order to load the transformer and then the text encoder. When it takes forever like this, it's because there's no VRAM or system RAM to load it into, so it loads it into swap instead! This can happen even if there is lots of system RAM available. On POP_OS the default "swappiness" is set to 180 which is super high and means the system tries to use swap very aggressively. I turned my swappiness right down so it only uses swap if it absolutely has to, and that makes the text encoder load quicker. However, if I push the training settings as high as they go, then it still hits swap and training can take 30 minutes to start. I can get it to train stably on videos or images at 768 resolution on that hardware, but it takes forever to start training.

I believe there are two bugs here that if solved would fix this. Number one is that when you're training with either unload text encoder or cache text embeddings then ai-toolkit is supposed to unload the text encoder since it's no longer needed. However, it first loads the transformer, so peak memory usage is higher than necessary because both the transformer and the text encoder are loaded even though the transformer isn't getting used to just prepare the embeddings. The second thing is is that when the text encoder actually does get unloaded, that doesn't actually free any memory!!!

These are two major bugs in ai-toolkit.

**Comment by MavenDE** [comment]
Look at VRAM usage. You need to have 5 or 6 GB of free VRAM for peeks. It can jump from 26GB to 30.5 GB even.
If you fill VRAM 80+% you have stall - nothing happens.

Example:
5090 with 31.5GB of free VRAM (a second GPU, never used for Windows).

Safe limit for WAN 2.2 is around 26-27GB of VRAM consumed for Ai Toolkit.

OFFLOAD + LOW VRAM + change HIGH and LOW every 10 steps if you train Wan 2.2.

LTX2 - Experiment with offloading % - start with 80% or so and go down till you leave some free GBs of VRAM like I said.

This traning needs hell lot of VRAM or RAM - not 64GB RAM and not 16GB VRAM. It won't work well.

My stats are for 1024/1280 pixels training.

Another thing - Look at the settings. Did you provide a valid path to diffusers directory for the model to load?
